tests/test_scripts.py

```diff
@@ -1926,3 +1926,15 @@ def test_readstats_empty():
     assert status == 0
 
     assert expected_output in out
+
+
+def test_existance_failure():
+    expected_output = 'ERROR: Input file'
+
+    args = ['thisfiledoesnotexistatall']
+
+    status, out, err = utils.runscript(
```

Comment:
This test should specifically test the changes you made to kfile.py - if you remove those changes, will this test still succeed? It doesn't look like it to me.

So, I think you should actually write two tests -- one test for a script like extract-paired-reads, that tests the error message and exit status.  And one test in test_script_arguments.py that runs the code in kfile directly, and verifies that behavior.  Note that sys.exit raises SystemExit, so you can look for that in the test.

(Let me know if you'd like me to expand on all of this!)